### PR TITLE
test: add test for FlightSQL CLI client

### LINKED ISSUE #3814: FlightSQL CLI client: simple integration test
Let's ave a simple integration test for the FlightSQL CLI client that connects to a server that returns 2 endpoints (both pointing to itself) and prints out the record batches. The setup can be used to test more upcoming features.

### PR DESCRIPTION
**Pull request by crepererum**
# Which issue does this PR close?
Closes #3814.

# Rationale for this change
Let's have some rough tests before adding more features.

# What changes are included in this PR?
- Tiny test w/ a dummy server.
- Fix for non-HTTPS connections.

# Are there any user-facing changes?
Using the CLI w/ non-HTTPS connections now works.

### CODE REVIEW

File: arrow-flight/tests/flight_sql_client_cli.rs [hunk 1]
```diff
@@ -0,0 +1,552 @@
...
+        \n| field_string | field_int |\
+        \n+--------------+-----------+\
+        \n| Hello        | 42        |\
+        \n| lovely       |           |\
+        \n| FlightSQL!   | 1337      |\
+        \n+--------------+-----------+",
+    );
+}
+
+const DEFAULT_TIMEOUT_SECONDS: u64 = 30;
```
Comment (alamb) [inline arrow-flight/tests/flight_sql_client_cli.rs:88]:
```suggestion
/// All tests must complete within this many seconds or else the test server is shutdown
const DEFAULT_TIMEOUT_SECONDS: u64 = 30;
```

File: arrow-flight/tests/flight_sql_client_cli.rs [hunk 2]
```diff
@@ -0,0 +1,552 @@
...
+
+        let mut string_builder = StringBuilder::new();
+        string_builder.append_value("Hello");
+        string_builder.append_value("lovely");
+        string_builder.append_value("FlightSQL!");
+
+        let mut int_builder = Int64Builder::new();
+        int_builder.append_value(42);
+        int_builder.append_null();
+        int_builder.append_value(1337);
```
Comment (alamb) [inline arrow-flight/tests/flight_sql_client_cli.rs:115]: You could perhaps use the more modern 

(untested)
```suggestion
        let string_array = StringArray::from(["Hello", "lovely", "FlightSQL!");
        let int_array = Int64Array::from([Some(42), None, Some(1337)]);
```

File: arrow-flight/tests/flight_sql_client_cli.rs [hunk 3]
```diff
@@ -0,0 +1,552 @@
...
+        &self,
+        _request: Request<Streaming<HandshakeRequest>>,
+    ) -> Result<
+        Response<Pin<Box<dyn Stream<Item = Result<HandshakeResponse, Status>> + Send>>>,
+        Status,
+    > {
+        Err(Status::unimplemented("do_handshake not implemented"))
+    }
+
+    async fn do_get_fallback(
```
Comment (alamb) [inline arrow-flight/tests/flight_sql_client_cli.rs:139]: This is awesome -- thank you @crepererum

**Review by alamb — APPROVED** [review]
Thank you @crepererum 

I ran the test  locally like

```
 cargo test --features=cli,tls,flight-sql-experimental --test flight_sql_client_cli
```

I also double checked that this test runs as part of CI(as it has several required features ) and it does ✅ 

https://github.com/apache/arrow-rs/actions/runs/4355722022/jobs/7612726842#step:6:141

cc @avantgardnerio

**Review by avantgardnerio — APPROVED** [review]
Love it, ty @crepererum

**Comment by alamb** [comment]
Thanks again @crepererum

**Pull request merged**